src/goats_tom/tasks/check_latest_version.py

```diff
@@ -0,0 +1,135 @@
+"""Check and publish the latest GOATS version info (WS + Redis cache)."""
+
+__all__ = ["check_version_info"]
+
+import logging
+from importlib.metadata import version as get_version
+from json import JSONDecodeError
+
+import dramatiq
+import requests
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+from django.core.cache import caches
+from packaging.version import InvalidVersion, Version
+
+logger = logging.getLogger(__name__)
+
+PACKAGE_NAME = "goats"
+CHANNELDATA_URL = "https://gemini-hlsw.github.io/goats-infra/conda/channeldata.json"
+GROUP_NAME = "updates_group"
+CACHE_KEY = "version:info"
+
+
+def _notify_version_info_ws(is_outdated: bool, current: str, latest: str) -> None:
```

Comment:
I started over in a different file. Thanks for laying the groundwork. The redis lock stuff is truly interesting! Glad I didn't have to figure it out. Nice job!